src/hooks/useStepsContext.ts

```diff
@@ -88,6 +96,108 @@ export function useStepsContext(): IStepsContext {
         return oldSteps;
       });
     },
+    /**
+     * Used in conjunction with the drag/drop functionality for
+     * moving a step separator in the UI to reorganize the bucketing
+     * of actions.
+     *
+     * @example
+     * The examples below depend on this array:
+     *
+     * [[1, 2, 3], [4, 5, 6], [7, 8, 9]]
+     *
+     *
+     * Example 1:
+     * Target index: 0
+     * Initiator index: 1
+     * Action index: 1
+     * Output: [[1, 2], [3, 4, 5, 6], [7, 8, 9]]
+     *
+     * Slice step 0 at idx 1, results in [1, 2]
+     * Slice step 0, starting at idx 2...length, results in [3]
+     * Merge [3] with step idx 1, [4, 5, 6]
+     *
+     * Example 2:
+     * Target index: 1
+     * Initiator index: 1
+     * Action index: 0
+     * Output: [[1, 2, 3, 4], [5, 6], [7, 8, 9]]
+     *
+     * Example 3:
+     * Target index: 2
+     * Initiator index: 1
+     * Action index: 0
+     * Output: [[1, 2, 3], [4, 5], [6, 7, 8, 9]]
+     */
```

Comment:
Thanks for adding this in, really happy to have such detailed examples and seeing the code refer back to them.